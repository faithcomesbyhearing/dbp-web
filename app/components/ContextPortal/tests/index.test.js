import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ContextPortal } from '..';

jest.mock('react-intl', () => ({
	FormattedMessage: ({ defaultMessage }) => <span>{defaultMessage}</span>,
	defineMessages: (messages) => messages,
}));

jest.mock(
	'../../PopupMessage',
	() =>
		function portalPopupMessageMock({ styles, message, x, y }) {
			return (
				<div
					style={{ top: y - 50, left: x - 87.5, ...styles }}
					className={'custom-popup'}
				>
					{message ? (
						<p>{message}</p>
					) : (
						<p>
							If you believe this to be an error please{' '}
							<a
								className={'logo'}
								href={'https://support.bible.is/contact'}
								title={'https://support.bible.is/contact'}
								target={'_blank'}
								rel={'noopener'}
							>
								contact support
							</a>
							.
						</p>
					)}
				</div>
			);
		},
);

const props = {
	selectedText: 'For all have sinned and fallen short of the glory of God.',
	coordinates: { x: 150, y: 150 },
	notesActive: false,
	isIe: false,
	addHighlight: jest.fn(),
	setActiveNote: jest.fn(),
	addFacebookLike: jest.fn(),
	toggleNotesModal: jest.fn(),
	closeContextMenu: jest.fn(),
	handleAddBookmark: jest.fn(),
	setActiveNotesView: jest.fn(),
};

describe('ContextPortal Component', () => {
	it('Should render and match snapshot', () => {
		const { container } = render(<ContextPortal {...props} />);
		expect(container).toMatchSnapshot();
	});

	it('Should render and match snapshot if IE is active', () => {
		const { container } = render(<ContextPortal {...props} isIe />);
		expect(container).toMatchSnapshot();
	});

	it('Should render and match snapshot if notes menu is active', () => {
		const { container } = render(<ContextPortal {...props} notesActive />);
		expect(container).toMatchSnapshot();
	});

	it('Should render and match snapshot if popup menu is active', () => {
		const { container } = render(<ContextPortal {...props} />);
		fireEvent.click(document); // Simulating the event that opens the popup
		expect(container).toMatchSnapshot();
	});

	it('Should render and match snapshot if highlight menu is open', () => {
		const { container } = render(<ContextPortal {...props} />);
		const highlightButton = container.querySelector('#add-highlight');
		fireEvent.click(highlightButton);
		expect(container).toMatchSnapshot();
	});

	it('Should handle click events correctly', () => {
		const { getByTitle } = render(<ContextPortal {...props} />);

		const notesButton = getByTitle('Add a note');
		fireEvent.click(notesButton);
		expect(props.setActiveNotesView).toHaveBeenCalledWith('edit');
		expect(props.toggleNotesModal).toHaveBeenCalled();
		expect(props.closeContextMenu).toHaveBeenCalled();

		const bookmarkButton = getByTitle('Add a bookmark');
		fireEvent.click(bookmarkButton);
		expect(props.setActiveNotesView).toHaveBeenCalledWith('bookmarks');
		expect(props.toggleNotesModal).toHaveBeenCalled();
		expect(props.handleAddBookmark).toHaveBeenCalled();
	});
});
