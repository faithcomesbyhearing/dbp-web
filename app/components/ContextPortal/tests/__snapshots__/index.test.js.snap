// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextPortal Component Should render and match snapshot 1`] = `
<div>
  <div
    class="context-menu shadow"
    style="position: absolute; left: 150px; top: 150px;"
  >
    <div
      class="menu-row"
    >
      <span
        class="menu-item normal"
        id="add-note"
        role="button"
        tabindex="0"
        title="Add a note"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#notes"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          Notes
        </span>
      </span>
      <span
        class="menu-item normal"
        id="add-highlight"
        role="button"
        tabindex="0"
        title="Add a highlight"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#highlight"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          Highlight
        </span>
        <div
          class="highlight-colors"
        >
          <span
            class="color-group"
          >
            <span
              class="none"
              role="button"
              tabindex="0"
            />
            <span
              class="color-text"
            >
              <span>
                None
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="yellow"
              role="button"
              tabindex="-1"
            />
            <span
              class="color-text"
            >
              <span>
                Yellow
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="green"
              role="button"
              tabindex="-2"
            />
            <span
              class="color-text"
            >
              <span>
                Green
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="pink"
              role="button"
              tabindex="-3"
            />
            <span
              class="color-text"
            >
              <span>
                Pink
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="purple"
              role="button"
              tabindex="-4"
            />
            <span
              class="color-text"
            >
              <span>
                Purple
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="blue"
              role="button"
              tabindex="-5"
            />
            <span
              class="color-text"
            >
              <span>
                Blue
              </span>
            </span>
          </span>
        </div>
      </span>
      <span
        class="menu-item normal"
        id="add-bookmark"
        role="button"
        tabindex="0"
        title="Add a bookmark"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#bookmark"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          Bookmark
        </span>
      </span>
      <button
        class="react-share__ShareButton menu-item normal email-share"
        id="email-share-button"
        style="background-color: transparent; padding: 0px; cursor: pointer;"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#e-mail"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          E-mail
        </span>
      </button>
    </div>
    <div
      class="menu-row"
    >
      <button
        class="react-share__ShareButton menu-item social facebook fb-share-button"
        id="facebook-share-button"
        quote=""For all have sinned and fallen short of the glory of God.""
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#facebook"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </button>
      <div
        class="SocialMediaShareButton SocialMediaShareButton--googlePlus menu-item social google"
        id="sharePost"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#google"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </div>
      <button
        class="react-share__ShareButton menu-item social twitter"
        id="twitter-share-button"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#twitter"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </button>
      <div
        class="menu-item social like-button"
        id="like-facebook"
        role="button"
        tabindex="0"
        title="Like on Facebook"
      >
        <span
          class="react-share__ShareCount"
        />
        <span
          class="like-thumb"
        >
          <svg
            height="24px"
            width="24px"
          >
            <use
              xlink:href="/svglist.svg#like_one-color"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            />
          </svg>
           
          Like
        </span>
      </div>
    </div>
    <div
      class="menu-row"
      id="copy-container"
      role="button"
      tabindex="0"
      title="Copy link to page"
    >
      <input
        class="copy-link"
        id="link-to-copy"
        readonly=""
        value="http://localhost/"
      />
      <span
        class="copy-button"
        id="copy-button"
      >
        Copy
      </span>
    </div>
  </div>
</div>
`;

exports[`ContextPortal Component Should render and match snapshot if IE is active 1`] = `
<div>
  <div
    class="context-menu shadow"
    style="position: absolute; left: 150px; top: 150px;"
  >
    <div
      id="internet-explorer-container"
    >
      <span>
        Bible.is does not support this feature on Internet Explorer. Please use one of the listed browsers instead:
      </span>
       
      <a
        href="https://www.google.com/chrome"
        target="_blank"
      >
        <span>
          Chrome
        </span>
      </a>
       
      <span>
        Or
      </span>
       
      <a
        href="https://www.mozilla.org/en-US/firefox/"
        target="_blank"
      >
        <span>
          Firefox
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`ContextPortal Component Should render and match snapshot if highlight menu is open 1`] = `
<div>
  <div
    class="context-menu shadow"
    style="position: absolute; left: 150px; top: 150px;"
  >
    <div
      class="menu-row"
    >
      <span
        class="menu-item normal"
        id="add-note"
        role="button"
        tabindex="0"
        title="Add a note"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#notes"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          Notes
        </span>
      </span>
      <span
        class="menu-item active"
        id="add-highlight"
        role="button"
        tabindex="0"
        title="Add a highlight"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#highlight"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          Highlight
        </span>
        <div
          class="highlight-colors active"
        >
          <span
            class="color-group"
          >
            <span
              class="none"
              role="button"
              tabindex="0"
            />
            <span
              class="color-text"
            >
              <span>
                None
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="yellow"
              role="button"
              tabindex="-1"
            />
            <span
              class="color-text"
            >
              <span>
                Yellow
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="green"
              role="button"
              tabindex="-2"
            />
            <span
              class="color-text"
            >
              <span>
                Green
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="pink"
              role="button"
              tabindex="-3"
            />
            <span
              class="color-text"
            >
              <span>
                Pink
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="purple"
              role="button"
              tabindex="-4"
            />
            <span
              class="color-text"
            >
              <span>
                Purple
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="blue"
              role="button"
              tabindex="-5"
            />
            <span
              class="color-text"
            >
              <span>
                Blue
              </span>
            </span>
          </span>
        </div>
      </span>
      <span
        class="menu-item normal"
        id="add-bookmark"
        role="button"
        tabindex="0"
        title="Add a bookmark"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#bookmark"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          Bookmark
        </span>
      </span>
      <button
        class="react-share__ShareButton menu-item normal email-share"
        id="email-share-button"
        style="background-color: transparent; padding: 0px; cursor: pointer;"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#e-mail"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          E-mail
        </span>
      </button>
    </div>
    <div
      class="menu-row"
    >
      <button
        class="react-share__ShareButton menu-item social facebook fb-share-button"
        id="facebook-share-button"
        quote=""For all have sinned and fallen short of the glory of God.""
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#facebook"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </button>
      <div
        class="SocialMediaShareButton SocialMediaShareButton--googlePlus menu-item social google"
        id="sharePost"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#google"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </div>
      <button
        class="react-share__ShareButton menu-item social twitter"
        id="twitter-share-button"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#twitter"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </button>
      <div
        class="menu-item social like-button"
        id="like-facebook"
        role="button"
        tabindex="0"
        title="Like on Facebook"
      >
        <span
          class="react-share__ShareCount"
        />
        <span
          class="like-thumb"
        >
          <svg
            height="24px"
            width="24px"
          >
            <use
              xlink:href="/svglist.svg#like_one-color"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            />
          </svg>
           
          Like
        </span>
      </div>
    </div>
    <div
      class="menu-row"
      id="copy-container"
      role="button"
      tabindex="0"
      title="Copy link to page"
    >
      <input
        class="copy-link"
        id="link-to-copy"
        readonly=""
        value="http://localhost/"
      />
      <span
        class="copy-button"
        id="copy-button"
      >
        Copy
      </span>
    </div>
  </div>
</div>
`;

exports[`ContextPortal Component Should render and match snapshot if notes menu is active 1`] = `
<div>
  <div
    class="context-menu shadow"
    style="position: absolute; left: 150px; top: 150px;"
  >
    <div
      class="menu-row"
    >
      <span
        class="menu-item normal"
        id="add-note"
        role="button"
        tabindex="0"
        title="Add a note"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#notes"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          Notes
        </span>
      </span>
      <span
        class="menu-item normal"
        id="add-highlight"
        role="button"
        tabindex="0"
        title="Add a highlight"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#highlight"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          Highlight
        </span>
        <div
          class="highlight-colors"
        >
          <span
            class="color-group"
          >
            <span
              class="none"
              role="button"
              tabindex="0"
            />
            <span
              class="color-text"
            >
              <span>
                None
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="yellow"
              role="button"
              tabindex="-1"
            />
            <span
              class="color-text"
            >
              <span>
                Yellow
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="green"
              role="button"
              tabindex="-2"
            />
            <span
              class="color-text"
            >
              <span>
                Green
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="pink"
              role="button"
              tabindex="-3"
            />
            <span
              class="color-text"
            >
              <span>
                Pink
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="purple"
              role="button"
              tabindex="-4"
            />
            <span
              class="color-text"
            >
              <span>
                Purple
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="blue"
              role="button"
              tabindex="-5"
            />
            <span
              class="color-text"
            >
              <span>
                Blue
              </span>
            </span>
          </span>
        </div>
      </span>
      <span
        class="menu-item normal"
        id="add-bookmark"
        role="button"
        tabindex="0"
        title="Add a bookmark"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#bookmark"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          Bookmark
        </span>
      </span>
      <button
        class="react-share__ShareButton menu-item normal email-share"
        id="email-share-button"
        style="background-color: transparent; padding: 0px; cursor: pointer;"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#e-mail"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          E-mail
        </span>
      </button>
    </div>
    <div
      class="menu-row"
    >
      <button
        class="react-share__ShareButton menu-item social facebook fb-share-button"
        id="facebook-share-button"
        quote=""For all have sinned and fallen short of the glory of God.""
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#facebook"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </button>
      <div
        class="SocialMediaShareButton SocialMediaShareButton--googlePlus menu-item social google"
        id="sharePost"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#google"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </div>
      <button
        class="react-share__ShareButton menu-item social twitter"
        id="twitter-share-button"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#twitter"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </button>
      <div
        class="menu-item social like-button"
        id="like-facebook"
        role="button"
        tabindex="0"
        title="Like on Facebook"
      >
        <span
          class="react-share__ShareCount"
        />
        <span
          class="like-thumb"
        >
          <svg
            height="24px"
            width="24px"
          >
            <use
              xlink:href="/svglist.svg#like_one-color"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            />
          </svg>
           
          Like
        </span>
      </div>
    </div>
    <div
      class="menu-row"
      id="copy-container"
      role="button"
      tabindex="0"
      title="Copy link to page"
    >
      <input
        class="copy-link"
        id="link-to-copy"
        readonly=""
        value="http://localhost/"
      />
      <span
        class="copy-button"
        id="copy-button"
      >
        Copy
      </span>
    </div>
  </div>
</div>
`;

exports[`ContextPortal Component Should render and match snapshot if popup menu is active 1`] = `
<div>
  <div
    class="context-menu shadow"
    style="position: absolute; left: 150px; top: 150px;"
  >
    <div
      class="menu-row"
    >
      <span
        class="menu-item normal"
        id="add-note"
        role="button"
        tabindex="0"
        title="Add a note"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#notes"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          Notes
        </span>
      </span>
      <span
        class="menu-item normal"
        id="add-highlight"
        role="button"
        tabindex="0"
        title="Add a highlight"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#highlight"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          Highlight
        </span>
        <div
          class="highlight-colors"
        >
          <span
            class="color-group"
          >
            <span
              class="none"
              role="button"
              tabindex="0"
            />
            <span
              class="color-text"
            >
              <span>
                None
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="yellow"
              role="button"
              tabindex="-1"
            />
            <span
              class="color-text"
            >
              <span>
                Yellow
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="green"
              role="button"
              tabindex="-2"
            />
            <span
              class="color-text"
            >
              <span>
                Green
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="pink"
              role="button"
              tabindex="-3"
            />
            <span
              class="color-text"
            >
              <span>
                Pink
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="purple"
              role="button"
              tabindex="-4"
            />
            <span
              class="color-text"
            >
              <span>
                Purple
              </span>
            </span>
          </span>
          <span
            class="color-group"
          >
            <span
              class="blue"
              role="button"
              tabindex="-5"
            />
            <span
              class="color-text"
            >
              <span>
                Blue
              </span>
            </span>
          </span>
        </div>
      </span>
      <span
        class="menu-item normal"
        id="add-bookmark"
        role="button"
        tabindex="0"
        title="Add a bookmark"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#bookmark"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          Bookmark
        </span>
      </span>
      <button
        class="react-share__ShareButton menu-item normal email-share"
        id="email-share-button"
        style="background-color: transparent; padding: 0px; cursor: pointer;"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#e-mail"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          class="item-text"
        >
          E-mail
        </span>
      </button>
    </div>
    <div
      class="menu-row"
    >
      <button
        class="react-share__ShareButton menu-item social facebook fb-share-button"
        id="facebook-share-button"
        quote=""For all have sinned and fallen short of the glory of God.""
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#facebook"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </button>
      <div
        class="SocialMediaShareButton SocialMediaShareButton--googlePlus menu-item social google"
        id="sharePost"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#google"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </div>
      <button
        class="react-share__ShareButton menu-item social twitter"
        id="twitter-share-button"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="/svglist.svg#twitter"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
      </button>
      <div
        class="menu-item social like-button"
        id="like-facebook"
        role="button"
        tabindex="0"
        title="Like on Facebook"
      >
        <span
          class="react-share__ShareCount"
        />
        <span
          class="like-thumb"
        >
          <svg
            height="24px"
            width="24px"
          >
            <use
              xlink:href="/svglist.svg#like_one-color"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            />
          </svg>
           
          Like
        </span>
      </div>
    </div>
    <div
      class="menu-row"
      id="copy-container"
      role="button"
      tabindex="0"
      title="Copy link to page"
    >
      <input
        class="copy-link"
        id="link-to-copy"
        readonly=""
        value="http://localhost/"
      />
      <span
        class="copy-button"
        id="copy-button"
      >
        Copy
      </span>
    </div>
  </div>
</div>
`;
